http/cors.go

```diff
@@ -0,0 +1,32 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+)
+
+func wrapCORSHandler(h http.Handler, core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		corsConf := core.CORSConfig()
```

Comment:
If you look at `cors.ApplyHeaders()` that is exactly what we do. If CORS isn't enable, then we just return and go on to the next handler in the chain. I do see that it might be helpful to move the logic to determine if CORS in enabled up to `wrapCORSHandler` just to make the code more readable. 

At the very least `Access-Control-Allow-Origin` needs to be returned with every cross-origin request. `Access-Control-Allow-Origin`, `Access-Control-Allow-Methods`, `Access-Control-Allow-Headers`, and `Access-Control-Max-Age` need to be on OPTIONS (preflight) requests, and that's what I'm doing.